packages/devtools_app/lib/src/provider/provider_screen.dart

```diff
@@ -0,0 +1,805 @@
+
+  return instanceDetails.maybeMap(
+    object: (instance) => instance.evalForInstance,
+    orElse: () => ref.watch(evalProvider),
+  );
+});
+
+class ProviderScreen extends Screen {
+  const ProviderScreen()
+      : super.conditional(
+          id: id,
+          requiresLibrary: 'package:provider/',
+          title: 'Provider',
+          icon: Icons.palette,
+        );
+
+  static const id = 'provider';
+
+  @override
+  Widget build(BuildContext context) {
+    return const ProviderScope(child: _ProviderScreenBody());
+  }
+}
+
+class _ProviderScreenBody extends ConsumerWidget {
+  const _ProviderScreenBody({Key key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context, ScopedReader watch) {
+    final splitAxis = Split.axisFor(context, 0.85);
+
+    return Split(
+      axis: splitAxis,
+      initialFractions: const [0.33, 0.67],
+      children: [
+        const _SplitBorder(child: _ProviderList()),
+        Column(
+          children: [
+            const Expanded(child: _SplitBorder(child: _ProviderValue())),
+            if (watch(_hasSelectedProviderProvider)) ...[
+              const SizedBox(height: 10),
+              const _SplitBorder(child: _ProviderEvaluation()),
+            ]
+          ],
+        ),
+      ],
+    );
+  }
+}
+
+void _showErrorSnackBar(BuildContext context, Object error) {
+  ScaffoldMessenger.of(context).showSnackBar(
+    SnackBar(content: Text('Error: $error')),
+  );
+}
+
+class _ProviderEvaluation extends StatefulWidget {
+  const _ProviderEvaluation({Key key}) : super(key: key);
+
+  @override
+  _ProviderEvaluationState createState() => _ProviderEvaluationState();
+}
+
+class _ProviderEvaluationState extends State<_ProviderEvaluation> {
+  final isAlive = IsAlive();
+
+  @override
+  void dispose() {
+    isAlive.dispose();
+    super.dispose();
+  }
+
+  Future<void> _evalExpression(
+    FutureOr<EvalOnDartLibrary> evalFuture,
+    String expression,
+  ) async {
+    try {
+      final eval = await evalFuture;
+
+      final selectedProviderId =
+          context.read(_selectedProviderIdProvider).state;
+      final providerInstance = await context.read(
+        instanceProvider(InstancePath.fromProvider(selectedProviderId)).future,
+      );
+
+      await eval.safeEval(
+        expression,
+        isAlive: isAlive,
+        scope: {
+          r'$value': providerInstance.instanceRefId,
+        },
+      );
+
+      unawaited(
+        context.refresh(
+          instanceProvider(InstancePath.fromProvider(selectedProviderId)),
+        ),
+      );
+
+      await serviceManager.performHotReload();
+    } catch (err) {
+      _showErrorSnackBar(context, err);
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.all(8.0),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          const Text('Execute code against the value exposed by this provider'),
```

Comment:
I've been playing around with a Console window exposed across most screens in DevTools. Similarly to the console in chrome devtools, the last N objects selected would be available as $0, $1, ... $6 in the console. Fine to land this separately for now but I'd be interested in unifying so there is one console experience across the app.